packages/x-data-grid-pro/src/hooks/features/dataSource/useGridDataSource.ts

```diff
@@ -52,27 +62,56 @@ export const useGridDataSource = (
     | 'sortingMode'
     | 'filterMode'
     | 'paginationMode'
+    | 'pageSizeOptions'
     | 'treeData'
+    | 'lazyLoading'
   >,
 ) => {
   const nestedDataManager = useLazyRef<NestedDataManager, void>(
     () => new NestedDataManager(apiRef),
   ).current;
+  const paginationModel = useGridSelector(apiRef, gridPaginationModelSelector);
   const groupsToAutoFetch = useGridSelector(apiRef, gridRowGroupsToFetchSelector);
+  const filteredSortedRowIds = useGridSelector(apiRef, gridFilteredSortedRowIdsSelector);
   const scheduledGroups = React.useRef<number>(0);
+
+  const isLazyLoaded = !!props.unstable_dataSource && props.lazyLoading;
   const onError = props.unstable_onDataSourceError;
 
+  const cacheChunkSize = React.useMemo(() => {
+    const sortedPageSizeOptions = props.pageSizeOptions
+      .map((option) => (typeof option === 'number' ? option : option.value))
+      .sort((a, b) => a - b);
+
+    return Math.min(paginationModel.pageSize, sortedPageSizeOptions[0]);
+  }, [paginationModel.pageSize, props.pageSizeOptions]);
+
   const [cache, setCache] = React.useState<GridDataSourceCache>(() =>
-    getCache(props.unstable_dataSourceCache),
+    getCache(props.unstable_dataSourceCache, {
+      chunkSize: cacheChunkSize,
+    }),
   );
 
   const fetchRows = React.useCallback(
-    async (parentId?: GridRowId) => {
+    async (options?: GridRowId | FetchRowsOptions) => {
```

Comment:
Could we do this instead? A few rationales:
1. Would not require deprecation of the current API.
2. Users won't have to worry about the options parameter for the non-lazyLoading use cases.
3. The signature looks more robust to me this way.
   - first/main argument = the parent for which the children are needed
   - second/supporting argument = override for the `GridGetRowsParams` (technically, the users may want to override any argument, so we may skip defining a dedicated `FetchRowsOptions` in favor of `Partial<GridGetRowsParams>`)
```suggestion
    async (parentId: GridRowId = GRID_ROOT_GROUP_ID, fetchParamsOverride?: Partial<GridGetRowsParams>) => {
```

Reply:
Another possibility that would remove the need to change this signature could be to handle the extraction of `start` and `end` in [`gridGetRowsParamsSelector`](https://github.com/mui/mui-x/blob/4bc5acc101590bf64eef33270ef803647e249dde/packages/x-data-grid-pro/src/hooks/features/dataSource/gridDataSourceSelector.ts#L17C14-L17C39) like we are computing them today, it may also add a smoother reactivity thanks to `useGridSelector` store binding.
https://github.com/mui/mui-x/blob/4bc5acc101590bf64eef33270ef803647e249dde/packages/x-data-grid-pro/src/hooks/features/dataSource/gridDataSourceSelector.ts#L29

Would that be feasible in this use-case?

Reply:
the point was to make it easier for users to call this via API once they face an error in the data fetch
error handler provides `GridGetRowsParams`, which then can be used to repeat the same request after user action
(used [here](https://deploy-preview-13878--material-ui-x.netlify.app/x/react-data-grid/server-side-data/lazy-loading/#error-handling))

I agree that `parentId` can be separated, since it is a bit more special than the rest. It should not affect the much the intention that I had

Reply:
`parentId` will probably be needed anyway to make the same for error handling of nested data lazy loading

Reply:
after checking this a bit further, I see two options to keep this clean:

1. keeping options as the only param
Even though lazy loading is the only scenario where these are needed, in other cases they are not needed at all (outside of the grid).

2. move `start` and `end` into state and use selector (like you mentioned). But that would have to be in the data source branch of the state since the request can span across multiple pages and page size would have to be constantly adjusted to produce proper values from
```
const computeStartEnd = (paginationModel: GridPaginationModel) => {
  const start = paginationModel.page * paginationModel.pageSize;
  const end = start + paginationModel.pageSize - 1;
  return { start, end };
};
```

Reason for not having two params is that the users would have to have this code in the fetch retry
`apiRef.current.unstable_dataSource.fetchRows(null, retryParams);`
Also, in our code, listener to the `'getRows'` event would be 
```
useGridApiEventHandler(
  apiRef,
  'getRows',
  runIf(isLazyLoaded, (params: GridGetRowsParams) => fetchRows(null, params)),
);
```

wdyt?

Reply:
> Reason for not having two params is that the users would have to have this code in the fetch retry
apiRef.current.unstable_dataSource.fetchRows(null, retryParams);

To me this looks cleaner: `apiRef.current.unstable_dataSource.fetchRows(GRID_ROOT_GROUP_ID, retryParams)`, it re-enforces the fact that all the current lazy-loaded children belong to the root group, the first parent of the grid. This, when combined with the nested lazy-loading should make some sense together:

Consider a tree-data use-case:
```tsx
// Retry on root level
apiRef.current.unstable_dataSource.fetchRows(GRID_ROOT_GROUP_ID, retryParams)
// Retry on a nested level
apiRef.current.unstable_dataSource.fetchRows(123, retryParams)

// For a regular root level refetch (no `lazyLoading`), they can skip `GRID_ROOT_GROUP_ID`
apiRef.current.unstable_dataSource.fetchRows()
```

Your points are valid too, and I'm not against a single param. It's more of a DX preference thing, and this one looks more cohesive to me.

Reply:
I will update the interface to have two params